apis/python/tests/test_dataframe.py

```diff
@@ -111,6 +111,53 @@ def test_dataframe(tmp_path, arrow_schema):
         assert not A.schema.allows_duplicates
 
 
+@pytest.mark.parametrize("appendable", [True, False])
```

Comment:
as noted in #956, this lacks coverage for the DataFrame indexing, specifically,  multiple indexing columns (and corresponding correct result from `shape`), and the full gamut of indexed column types (eg., datetime, float, etc).

Reply:
At a minimum, please add tests related to shape: ie., that if a dataframe has >1 index columns, it generates the right shape

Reply:
As of the most recent commit, this PR retracts any attempts at explicit shaping of DataFrame. Those questions will now be addressed on a separate PR.

The asks here will not be forgotten; they are tracked on #961.

Reply:
Action plan here:
https://github.com/single-cell-data/TileDB-SOMA/pull/952#issuecomment-1434690695